site/content/docs/02-template-syntax.md

```diff
@@ -0,0 +1,592 @@
+
+* `{@html expression}`
+
+---
+
+In a text expression, characters like `<` and `>` are escaped. With HTML expressions, they're not.
+
+> Svelte does not sanitize expressions before injecting HTML. If the data comes from an untrusted source, you must sanitize it, or you are exposing your users to an XSS vulnerability.
+
+```html
+<div class="blog-post">
+	<h1>{post.title}</h1>
+	{@html post.content}
+</div>
+```
+
+
+### If blocks
+
+* `{#if expression}...{/if}`
+* `{#if expression}...{:else if expression}...{/if}`
+* `{#if expression}...{:else}...{/if}`
+
+---
+
+Content that is conditionally rendered can be wrapped in an if block.
+
+```html
+{#if answer === 42}
+	<p>what was the question?</p>
+{/if}
+```
+
+---
+
+Additional conditions can be added with `{:else if expression}`, optionally ending in an `{:else}` clause.
+
+```html
+{#if porridge.temperature > 100}
+	<p>too hot!</p>
+{:else if 80 > porridge.temperature}
+	<p>too cold!</p>
+{:else}
+	<p>just right!</p>
+{/if}
+```
+
+
+### Each blocks
+
+* `{#each expression as name}...{/each}`
+* `{#each expression as name, index}...{/each}`
+* `{#each expression as name, index (key)}...{/each}`
+* `{#each expression as name}...{:else}...{/each}`
+
+---
+
+Iterating over lists of values can be done with an each block.
+
+```html
+<h1>Shopping list</h1>
+<ul>
+	{#each items as item}
+		<li>{item.name} x {item.qty}</li>
+	{/each}
+</ul>
+```
+
+---
+
+An each block can also specify an *index*, equivalent to the second argument in an `array.map(...)` callback:
+
+```html
+{#each items as item, i}
+	<li>{i + 1}: {item.name} x {item.qty}</li>
+{/each}
+```
+
+---
+
+If a *key* expression is provided, Svelte will diff the list when data changes, rather than adding or removing items at the end.
+
+```html
+{#each items as item, i (item.id)}
+	<li>{i + 1}: {item.name} x {item.qty}</li>
+{/each}
+```
+
+---
+
+You can freely use destructuring patterns in each blocks.
+
+```html
+{#each items as { id, name, qty }, i (id)}
+	<li>{i + 1}: {name} x {qty}</li>
+{/each}
+```
+
+---
+
+An each block can also have an `{:else}` clause, which is rendered if the list is empty.
+
+```html
+{#each todos as todo}
+	<p>{todo.text}</p>
+{:else}
+	<p>No tasks today!</p>
+{/each}
+```
+
+
+### Await blocks
+
+* `{#await expression}...{:then name}...{:catch name}...{/await}`
+* `{#await expression}...{:then name}...{/await}`
+* `{#await expression then name}...{/await}`
+
+---
+
+Await blocks allow you to branch on the three possible states of a Promise — pending, fulfilled or rejected.
+
+```html
+{#await promise}
+	<!-- promise is pending -->
+	<p>waiting for the promise to resolve...</p>
+{:then value}
+	<!-- promise was fulfilled -->
+	<p>The value is {value}</p>
+{:catch error}
+	<!-- promise was rejected -->
+	<p>Something went wrong: {error.message}</p>
+{/await}
+```
+
+---
+
+The `catch` block can be omitted if no error is possible, and the initial block can be omitted if you don't care about the pending state.
```

Comment:
I think rather than talking about no error being possible or about not caring about the pending state, we should just say that the await block renders as nothing in those cases if those sections are left out.

Reply:
👍